### PR TITLE
test(s3/lifecycle): bundle dispatcher + engine edge-case coverage

### PR DESCRIPTION
**Pull request by chrislusf**
Two-package bundle covering uncovered branches in production code that the existing happy-path tests don't reach. Dispatcher 58.1% → 60.2% and engine 81.0% → 81.7% (engine lift modest because most branches were already hit; the nil-rule defensive case is otherwise unreachable from a Compile flow).

**Dispatcher (6 tests):**
- \`FilerPersister.Load\` with nil Store errors with a "nil Store" message rather than panicking at the Read call
- \`FilerPersister.Save\` with nil Store same
- \`FilerPersister.Load\` with a non-NotFound transport error wraps the shard ID into the message AND keeps the underlying error recoverable via \`errors.Is\`
- \`FilerPersister.Load\` with successful empty \`[]byte\` returns an empty map, not a JSON-decode error — pinning that an existing-but-empty cursor file is treated as "no entries"
- \`Tick\` initializes the \`retries\` map on first call without panic so a freshly-constructed Dispatcher works
- \`Tick\` with already-canceled ctx re-queues the popped Match, returns zero, and never invokes the LifecycleDelete client — the Match must not be lost across worker restart

**Engine (4 tests):**
- \`rulePredicateSensitive(nil)\` returns false rather than panicking on the FilterTags dereference. Non-nil paths run through Compile, but a defensive nil-rule arrival isn't reachable that way
- rule with no FilterTags / empty FilterTags map returns false (the check is \`len(FilterTags) > 0\`, so empty must classify as non-sensitive — pinning catches a flipped \`>=\` comparison)
- rule with populated FilterTags returns true

## Summary by CodeRabbit

* **Bug Fixes**
  * Enhanced S3 lifecycle dispatcher to properly re-queue all remaining tasks during shutdown, improving reliability and preventing potential task loss.

* **Tests**
  * Added comprehensive edge-case test coverage for dispatcher shutdown behavior, persistence layer error handling, and rule filtering logic.

[Review Change Stack](https://app.coderabbit.ai/change-stack/seaweedfs/seaweedfs/pull/9413)

### CODE REVIEW

File: weed/s3api/s3lifecycle/dispatcher/edge_cases_test.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+	})
+
+	ctx, cancel := context.WithCancel(context.Background())
+	cancel() // shutdown before Tick even starts dispatching
+
+	processed := d.Tick(ctx, t0.Add(time.Hour))
+	assert.Equal(t, 0, processed, "shutdown must report zero processed")
+	assert.Equal(t, 0, calls, "shutdown must skip the dispatch call")
+	assert.Equal(t, 1, sched.Len(), "the popped Match must be re-queued")
+}
```
[the remark this answers is not in the document]
Comment (chrislusf) [reply weed/s3api/s3lifecycle/dispatcher/edge_cases_test.go:134]: Already addressed in 6965e26b3 — `Tick` re-queues every drained Match (`due[i:]`) on shutdown, and the test now enqueues three due Matches with `assert.Equal(t, 3, sched.Len())`. The bug you describe was real; the fix is in the latest commit.
